src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/magical/talismans/Talisman.java

```diff
@@ -179,6 +183,53 @@ public static boolean checkFor(Event e, SlimefunItem item) {
         }
     }
 
+    @Nullable
+    public static ItemStack checkForAndGet(@Nonnull SlimefunItemStack stack, @Nonnull Player player) {
+        return checkForAndGet(stack.getItem(), player);
+    }
+
+    @Nullable
+    public static ItemStack checkForAndGet(@Nonnull SlimefunItem sfItem, @Nonnull Player player) {
```

Comment:
This basically does the same as `checkFor` other than having a different return type right?
If so, then one of these is redundant, you could also just rewrite `checkFor` to basically run this method and do an != null check on the returned ItemStack.

Reply:
Its actually quite different, checkFor also consumes the item, adds potion effects and all that, checkForAndGet just checks for the talisman and returns it